fpptatweaks.php

```diff
@@ -514,47 +514,47 @@ function fpptatweaks_civicrm_searchColumns( $objectName, &$headers,  &$rows, &$s
       // ..when injecting the js for the value
       $insertHeaderPosition = 5;
       array_splice( $headers, $insertHeaderPosition, 0, [$insertedHeader] );
+      CRM_Core_Resources::singleton()->addVars('columnPosition', [$insertHeaderPosition]);
       // Add weight on the header for the final position of the columns
-      foreach ($headers as $key => $head) {
-        $headers[$key]['weight'] = $key;
+      foreach ($headers as $headerKey => $header) {
+        $headers[$headerKey]['weight'] = $headerKey;
       }
 
       // Get participantIds array of column data
-      $participantIds = CRM_Utils_Array::collect('participant_id', $rows);
-      $contributionIds = [];
+      $participantIdsPerRow = CRM_Utils_Array::collect('participant_id', $rows);
+      $contributionIdsPerRow = [];
       // Loop participantIds array to get contribution ID
-      foreach ($participantIds as $key => $participantId) {
+      foreach ($participantIdsPerRow as $rowKey => $participantId) {
         $paymentDetails = civicrm_api3('ParticipantPayment', 'get', [
           'sequential' => 1,
           'participant_id' => $participantId,
         ]);
 
         // Save contributionIds as array
-        $contributionIds[$key] = $paymentDetails['values'][0]['contribution_id'];
+        $contributionIdsPerRow[$rowKey] = $paymentDetails['values'][0]['contribution_id'];
       }
 
       // Flip for the rows position
-      $keyPerContributionId = array_flip($contributionIds);
+      $rowKeyPerContributionId = array_flip($contributionIdsPerRow);
     }
     else {
       $headers[] = $insertedHeader;
-      $contributionIds = CRM_Utils_Array::collect('contribution_id', $rows);
-      $keyPerContributionId = array_flip($contributionIds);
+      $contributionIdsPerRow = CRM_Utils_Array::collect('contribution_id', $rows);
+      $rowKeyPerContributionId = array_flip($contributionIdsPerRow);
     }
 
     $contributions = \Civi\Api4\Contribution::get()
       ->addSelect('id', 'invoice_number')
-      ->addWhere('id', 'IN', $contributionIds)
+      ->addWhere('id', 'IN', $contributionIdsPerRow)
       ->execute();
 
-    foreach ($contributions as $key => $contribution) {
-      $rows[$keyPerContributionId[$contribution['id']]]['invoice_number'] = $contribution['invoice_number'];
+    foreach ($contributions as $contribution) {
+      $rows[$rowKeyPerContributionId[$contribution['id']]]['invoice_number'] = $contribution['invoice_number'];
     }
 
     if ($objectName == 'event') {
       // Inject data in js
       CRM_Core_Resources::singleton()->addVars('eventRows', $rows);
```

Comment:
Oops, I should have mentioned this earlier:

When using `CRM_Core_Resources::singleton()->addVars(`, please always wrap all variables in an array that's named specifically to the extension. As written now, this code will store values in JS objects `CRM.vars.eventRows` and `CRM.vars.columnPosition`. Instead, we should avoid storing our variables directly in `CRM.vars` and  store these as `CRM.vars.fpptatweaks.columnPosition` and `CRM.vars.fpptatweaks.eventRows`.  Can you make that change please @lucky091588